myuw/views/api/instructor_section.py

```diff
@@ -78,6 +80,27 @@ def make_http_resp(self, timer, request, section_id):
         resp_data = load_schedule(request, schedule,
                                   section_callback=self.per_section_data)
 
+        _set_current_or_future(self.term, request, resp_data)
+
+        # Concurrently fetch section data and ICD data
+        threads = []
+
+        t = Thread(target=self.add_linked_section_data,
+                   args=(resp_data,))
+        threads.append(t)
+        t.start()
+
+        for section in resp_data['sections']:
```

Comment:
Instead of adding the for loop here, can we move this to per_section_data function?